packages/react-core/src/demos/examples/Nav/NavDefault.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import {
+  Card,
+  CardBody,
+  Gallery,
+  GalleryItem,
+  Nav,
+  NavItem,
+  NavList,
+  Page,
+  PageSection,
+  PageSectionVariants,
+  PageSidebar,
+  PageSidebarBody,
+  SkipToContent,
+  TextContent,
+  Text
+} from '@patternfly/react-core';
+import { DashboardBreadcrumb } from '../DashboardWrapper';
+import DashboardHeader from '../DashboardHeader';
+
+export const PageLayoutDefaultNav: React.FunctionComponent = () => {
+  const [activeItem, setActiveItem] = React.useState<number | string>(0);
+
+  const onNavSelect = (
+    _event: React.FormEvent<HTMLInputElement>,
+    selectedItem: {
+      groupId: number | string;
+      itemId: number | string;
+      to: string;
+    }
+  ) => setActiveItem(selectedItem.itemId);
+
+  const PageNav = (
+    <Nav onSelect={onNavSelect} aria-label="Nav">
+      <NavList>
+        <NavItem itemId={0} isActive={activeItem === 0} to="#">
```

Comment:
Can we just pass in the `preventDefault` prop on NavItem in these demos, and provide a single comment in each demo to state "Preventing default click behavior for demo purposes only"? Right now the activeItem state is rather unnecessary since it just gets reset when clicking a nav item.

Reply:
Good idea, this will solve the issue which occurs, after clicking on a NavItem, as it does not change the active item immediately, but after getting `#` on the URL path.

So just to be clear, after clicking on a NavItem, it adds the `to` anchor to the URL path, which resets the state, and then the UI works as expected, because once we are at `navigation/react-demos/default-nav#` (and all other NavItems also lead to `#`), the state is kept, as we are still on the same URL.

And by using `preventDefault /* If true prevents the default anchor link action to occur */` it does not do the link change.